### ISSUE TITLE
Use wrapt

### ISSUE THREAD

**Issue by interrogator** [bitpanda-labs/loggo#49]
the `wrapt` module provides some helper decorators for making class decorators that attach to all methods. we do some creepy stuff to make this work, so it'd be better to rely on a well-known implementation.

**Comment by interrogator** [comment]
So last time we looked at this, we realised it's actually a bit hard to do, though it likely is the best way forward. I won't be even attempting this in the near future, but we can leave this open.

**Comment by hukkinj1** [comment]
Yeah I tried once (unfortunately didnt seem to push the code anywhere so can't find it now). Had an implementation but still had some of the tests failing. So yes, this might be harder than it sounds.
